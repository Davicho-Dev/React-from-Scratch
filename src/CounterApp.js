import PropTypes from 'prop-types'
import { useState } from 'react'

export const CounterApp = ( { value } ) => {
	const [ counter, setCounter ] = useState( value )

	const hdlAdd = () => setCounter( counter + 1 )

	const hdlSubtract = () => setCounter( counter - 1 )

	const hdlReset = () => setCounter( value )

	return <>
		<h1>CounterApp</h1>
		<hr/>
		<h2>{ counter }</h2>
		<button onClick={ hdlAdd }>+1</button>
		<button onClick={ hdlReset }>Reset</button>
		<button onClick={ hdlSubtract }>-1</button>
	</>
}

CounterApp.propTypes = {
	value: PropTypes.number,
}

CounterApp.defaultProps = {
	value: 10
}
