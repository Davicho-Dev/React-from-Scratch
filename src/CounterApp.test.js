import { shallow } from 'enzyme'

import { CounterApp } from './CounterApp'

describe( 'Test in CounterApp', () => {
	test( 'It must display <CounterApp />correctly.', () => {
		const wrapper = shallow( <CounterApp/> )

		expect( wrapper ).toMatchSnapshot()
	} )

	test( '', () => {
		const wrapper = shallow( <CounterApp value={ 100 }/> )

		expect( wrapper.find( 'h2' ).text() ).toBe( '100' )
	} )
} )
