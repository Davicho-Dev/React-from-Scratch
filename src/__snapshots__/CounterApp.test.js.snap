// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in CounterApp It must display <CounterApp />correctly. 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <hr />
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
